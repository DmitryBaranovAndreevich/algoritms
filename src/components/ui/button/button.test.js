import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

const testOnClick = () => {
  alert('Click на кнопку');
}


it("Пустая кнопка рендерится без ошибок", () => {
  const emptyButton = renderer.create(<Button></Button>).toJSON();
  expect(emptyButton).toMatchSnapshot();
});

it("Кнопка с текстом рендериться без ошибок", () => {
  const textButton = renderer.create(<Button text="Нажми меня"></Button>).toJSON();
  expect(textButton).toMatchSnapshot();
});

it("Заблокированная кнопка рендериться без ошибок", () => {
  const disabledButton = renderer
    .create(<Button text="Нажми меня" disabled="true"></Button>)
    .toJSON();
  expect(disabledButton).toMatchSnapshot();
});

it("Кнопка с loader рендериться без ошибок", () => {
  const loaderButton = renderer
    .create(<Button text="Нажми меня" isLoader="true"></Button>)
    .toJSON();
  expect(loaderButton).toMatchSnapshot();
});

it("Нажатие на кнопку вызывает корректный alert", () => {
  window.alert = jest.fn();

  render(<Button text="Кликни меня" onClick={testOnClick}></Button>);

  const button = screen.getByText("Кликни меня");
  fireEvent.click(button);

  expect(window.alert).toHaveBeenCalledWith("Click на кнопку");
});