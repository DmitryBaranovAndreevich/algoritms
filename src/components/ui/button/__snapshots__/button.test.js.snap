// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Заблокированная кнопка рендериться без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  disabled="true"
  type="button"
>
  <p
    className="text undefined"
  >
    Нажми меня
  </p>
</button>
`;

exports[`Кнопка с loader рендериться без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  data-testid="button"
  disabled="true"
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Кнопка с текстом рендериться без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    className="text undefined"
  >
    Нажми меня
  </p>
</button>
`;

exports[`Пустая кнопка рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
